frontend/src/component/common/MessageBanner/MessageBanner.tsx

```diff
@@ -72,44 +118,64 @@ export const MessageBanner = () => {
             <StyledIcon variant={variant}>
                 <BannerIcon icon={icon} variant={variant} />
             </StyledIcon>
-            <StyledMessage dangerouslySetInnerHTML={{ __html: message }} />
+            <StyledMessage>
+                <ReactMarkdown>{message}</ReactMarkdown>
+            </StyledMessage>
             <BannerButton
                 link={link}
                 linkText={linkText}
                 plausibleEvent={plausibleEvent}
+                openDialog={() => setOpen(true)}
             />
+            <MessageBannerDialog
+                open={open}
+                setOpen={setOpen}
+                title={dialogTitle || linkText}
+            >
+                {dialog!}
+            </MessageBannerDialog>
         </StyledBar>
     );
 };
 
+const VariantIcons = {
+    warning: <WarningAmber />,
+    info: <InfoOutlined />,
+    error: <Close />,
+    success: <Check />,
+    neutral: <InfoOutlined />,
+    secondary: <InfoOutlined />,
+};
```

Comment:
Maybe @NicolaeUnleash can help us align the icons in the future, mapping the correct ones to each variant.